src/local/database.rs

```diff
@@ -19,35 +18,37 @@ pub struct SqliteDatabase {
     conn: Connection,
 }
 
-impl SqliteDatabase{
-    pub fn new(path: &str) -> Result<SqliteDatabase> {
-        let path = shellexpand::full(path)?;
+impl SqliteDatabase {
+    pub fn new(path: impl AsRef<Path>) -> Result<SqliteDatabase> {
         let path = path.as_ref();
-
         debug!("opening sqlite database at {:?}", path);
 
-        let create = !Path::new(path).exists();
+        let create = !path.exists();
+        if create {
+            if let Some(dir) = path.parent() {
+                std::fs::create_dir_all(dir)?;
+            }
+        }
```

Comment:
Turns out the `setup_db` function below doesn't create the dir, so this is necessary.

Reply:
makes sense